lib/LSM9DS1/LSM9DS1.cpp

```diff
@@ -125,6 +142,154 @@ void LSM9DS1::readMagnetosensor(float &x, float &y, float &z) {
     z = z_raw * resolution / 1000;
 }
 
+void LSM9DS1::configXLInterrupt(INT_XL_CONFIG intConfig, bool andInterrupt, uint8_t duration) {
```

Comment:
Read the configuration in the registers here and update those values instead of writing directly to the register to avoid overwriting bits that were already set